lute/templates/read/index.html

```diff
@@ -529,6 +529,39 @@ <h2>&#127881;</h2>
     });
   }
 
+  function add_bookmark() {
+    const text_id = document.querySelector('#text_id')?.value;
+    const book_id = document.querySelector('#book_id')?.value;
+    const page_num = document.querySelector('#page_num')?.value;
+    let title = prompt(`Enter a bookmark title for page ${page_num}`);
+
+    if (!book_id || !page_num || !title) {
+      alert(`Missing bookmark data. No bookmark added.`);
+      return
+    }
+
+    fetch(`/bookmarks/add`, {
+      method: "POST",
+      cache: "no-cache",
+      credentials: "same-origin",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify({
+        title: title,
+        book_id: book_id,
+        page_num: page_num,
+      })
+    }).then((resp) => resp.json()).then(json => {
+      if (json?.success === true) {
+        alert("Bookmark added!")
+      } else {
+        alert(`Unable to add bookmark. ${json?.reason || "Unknown reason"}`)
+      }
+    })
+
+    // TODO: notify user that bookmark was added
```

Comment:
Great point, but maybe the alert is sufficient.  Were you thinking of a flash message, or some kind of icon?

Reply:
Yeah, I wasn't sure if you'd want a nicer notification like a flash message or something. I think the `alert` works well enough for now, but I just put it in there as a placeholder.

If the `alert` is good enough for you though, want me to just remove the TODO?